skopt/plots.py

```diff
@@ -88,3 +89,65 @@ def plot_convergence(*args, **kwargs):
         ax.legend(loc="best")
 
     return ax
+
+
+def plot_scatter_matrix(result, bins=20):
+    """Plot scatter plot matrix of samples.
+
+    The diagonal shows a histogram for each dimension. Create pairwise
+    scatter plots of each dimension of the search space. Later
+    samples are shown in a darker color.
+
+    Parameters
+    ----------
+    * `result` [`OptimizeResult`]
+        The result for which to plot the scatter plot matrix.
+
+    * `bins` [int, bins=20]:
+        Number of bins to use for histograms on the diagonal.
+
+    Returns
+    -------
+    * `ax`: [`Axes`]:
+        The matplotlib axes.
+    """
+    samples = np.asarray(result.x_iters)
```

Comment:
Shall we document that this function cannot be used for `Categorical` dimensions?